src/persist-client/src/write.rs

```diff
@@ -112,11 +112,6 @@ where
     ///
     /// The clunky multi-level Result is to enable more obvious error handling
     /// in the caller. See <http://sled.rs/errors.html> for details.
-    ///
-    /// TODO: Introduce an AsyncIterator (futures::Stream) variant of this. Or,
```

Comment:
aljoscha's PR to split the API into batch and then commit addresses this need IMO